src/Features/Core/Portable/ExtractMethod/MethodExtractor.cs

```diff
@@ -35,8 +36,8 @@ internal abstract partial class MethodExtractor
             LocalFunction = localFunction;
         }
 
-        protected abstract Task<AnalyzerResult> AnalyzeAsync(SelectionResult selectionResult, bool localFunction, CancellationToken cancellationToken);
-        protected abstract Task<InsertionPoint> GetInsertionPointAsync(SemanticDocument document, CancellationToken cancellationToken);
+        protected abstract AnalyzerResult Analyze(SelectionResult selectionResult, bool localFunction, CancellationToken cancellationToken);
```

Comment:
synchronous, and now operates on the *original* user document/semantic model  so no forking/generaotrs that run.